src/components/learner-credit-management/data/hooks/useBudgetDetailPage.js

```diff
@@ -0,0 +1,76 @@
+import { getBudgetStatus } from '../utils';
+import { BUDGET_STATUSES } from '../../../EnterpriseApp/data/constants';
+
+const transformOfferToPolicy = (offer) => {
+  if (!offer) { return null; }
+
+  return {
+    displayName: null,
+    subsidyActiveDatetime: offer.dateCreated,
+    subsidyExpirationDatetime: offer.exportTimestamp,
```

Comment:
As discussed in Slack, the metadata needed about the offer regarding start/end dates are not available to this page route at the moment. We will need to integrate with the `enterprise-admin-offers` detail API to request the metadata about the currently viewed enterprise offer from ecommerce to get these dates (and the display name for the offer). I'd recommend implementing a simple `useEnterpriseOffer` hook for the budget detail page that makes the API request using `useQuery`.

Example:

```jsx
const useEnterpriseOffer = (enterpriseOfferId) => {
  return useQuery({
    queryKey: learnerCreditManagementQueryKeys.budgetEnterpriseOffer(enterpriseOfferId),
    queryFn: fetchEnterpriseOffer, // responsible for actually calling the ecommerce `enterprise-admin-offers` detail REST API, and camelCasing the response data.
    enabled: !!enterpriseOfferId, // keep the query disabled unless we are dealing with an enterprise offer
  });
};
```

See [Slack message](https://twou.slack.com/archives/D066M6FC0D7/p1701809304543899) from our DM for more context.